src/webhook/mutation/pod_mutator/init_container.go

```diff
@@ -27,22 +28,39 @@ func createInstallInitContainerBase(webhookImage, clusterID string, pod *corev1.
 			{Name: config.K8sNamespaceEnv, ValueFrom: kubeobjects.NewEnvVarSourceForField("metadata.namespace")},
 			{Name: config.K8sNodeNameEnv, ValueFrom: kubeobjects.NewEnvVarSourceForField("spec.nodeName")},
 		},
-		SecurityContext: copyUserContainerSecurityContext(pod),
+		SecurityContext: securityContextForInitContainer(pod),
 		Resources:       *dynakube.InitResources(),
 	}
 }
 
-func copyUserContainerSecurityContext(pod *corev1.Pod) *corev1.SecurityContext {
-	var securityContext *corev1.SecurityContext
-	if len(pod.Spec.Containers) == 0 {
-		return securityContext
-	}
+func securityContextForInitContainer(pod *corev1.Pod) *corev1.SecurityContext {
+	var securityContext = new(corev1.SecurityContext)
+
 	if pod.Spec.Containers[0].SecurityContext != nil {
-		securityContext = pod.Spec.Containers[0].SecurityContext.DeepCopy()
+		securityContext.RunAsGroup = pod.Spec.Containers[0].SecurityContext.RunAsGroup
+		securityContext.RunAsUser = pod.Spec.Containers[0].SecurityContext.RunAsUser
+	} else {
+		securityContext.RunAsGroup = address.Of(int64(1001))
+		securityContext.RunAsUser = address.Of(int64(1001))
 	}
+
+	limitSecurityContext(securityContext)
+
 	return securityContext
 }
 
+func limitSecurityContext(ctx *corev1.SecurityContext) {
+	ctx.RunAsNonRoot = address.Of(true)
```

Comment:
```yaml
seccompProfile:
              type: RuntimeDefault
```
seccomp profile can also be limited, see [k8s docs](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-seccomp-profile-for-a-container)

Reply:
done